lib/ansible/executor/task_result.py

```diff
@@ -132,10 +132,10 @@ def is_skipped(self) -> bool:
         return bool(self._return_data.get('skipped', False))
 
     def is_failed(self) -> bool:
-        if 'failed_when_result' in self._return_data or any(isinstance(loop_res, dict) and 'failed_when_result' in loop_res for loop_res in self._loop_results):
-            return self._check_key('failed_when_result')
+        if (value := self._return_data.get('failed_when_result', self._return_data.get('failed', ...))) is not ...:
+            return bool(value)
 
-        return self._check_key('failed')
+        return any(isinstance(result, dict) and result.get('failed_when_result', result.get('failed')) for result in self._loop_results)
```

Comment:
Try structuring like this:
```suggestion
        return any(
            result.get('failed_when_result', result.get('failed'))
            for result in self._loop_results
            if isinstance(result, dict)
        )
```